### ISSUE TITLE
Type propagation doesn't work for map keys

### ISSUE THREAD

**Issue by ajor** [bpftrace/bpftrace#3448]
This script:
```
BEGIN
{
  @stacks[@integer] = kstack;
  @integer = 1;
}
```
Gives an error:
```
stdin:3:3-20: ERROR: Argument mismatch for @stacks: trying to access with arguments: [uint64] when map expects arguments: [none]
  @stacks[@integer] = kstack;
  ~~~~~~~~~~~~~~~~~
```

The type of the `@integer` map should be back-propagated, making the original script roughly equivalent to this:
```
BEGIN
{
  @integer = 0;
  @stacks[@integer] = kstack;
  @integer = 1;
}
```

**Comment by ajor** [comment]
Maybe related, this script crashes bpftrace with a "Floating point exception":
```
BEGIN
{
  @map1[@map2] = 1;
  @map2 = 1;

  for ($kv : @map1) {
  }
}
```

Please check if this second script no longer crashes once the type propagation issue is fixed.

**Comment by ajor** [comment]
Recommend leaving this until after #3390 is merged as there will be conflicts.

**Comment by viktormalik** [comment]
@jordalgo is this resolved by #3536?

**Comment by jordalgo** [comment]
Ah yes. Thanks for the follow up.
